### ISSUE TITLE
Dimensions missing in druid

### ISSUE THREAD

**Issue by salsakran** [metabase/metabase#2376]
"I integrated Metabase with Druid Datasources. Unfortunately, Metabase doesn't show my Dimensions. It only shows Metric fields. Is this by design?

To show a sample of what I am talking, here is a snapshot from Pivot [another Druid Visualizer from imply.io]: Notice the Dimension fields on the left hand side (which are used for group by). Metabase doesn't show any dimensions."

From http://discourse.metabase.com/t/druid-metabase-does-not-shown-dimensions/333

![](http://discourse.metabase.com/uploads/default/original/1X/ea46ba0ee7fe92d8a547ba1fa26e30a86632429b.png)

**Comment by skilledmonster** [comment]
While this is fixed in the current version, is there a previous version that showed dimensions? If yes, I would look like explore that version and continue my evaluation. Thanks!

**Comment by salsakran** [comment]
@skilledmonster I'm a little confused. In all our tests, we see dimensions in the filter + group by selectors.

Can you check the data model page (and perhaps screen cap it) and the filter/grouping widgets for the dimensions?

**Comment by skilledmonster** [comment]
After the Druid is connected to the Broker, Metabase was able to pull up the datasources (tables) available -
![image](https://cloud.githubusercontent.com/assets/2034585/14649090/83a15780-0633-11e6-9a0f-3154cf22b03a.png)

"Filtered By" section shows metrics (as defined in druid datasource) - 
![image](https://cloud.githubusercontent.com/assets/2034585/14649185/ea287254-0633-11e6-8f3b-790159e89b21.png)

Under "View" Add a grouping ALSO shows metrics - 
![image](https://cloud.githubusercontent.com/assets/2034585/14649239/2317a4f4-0634-11e6-95d7-1f8985033f4d.png)

I don't see dimensions at all.

**Comment by agilliland** [comment]
@skilledmonster , if indeed there are columns missing from your tables then it's likely something happened when the database was being scanned.  can you check your application log files and see if anything shows up there?  in particular you'll want to try syncing your database and then checking out the logs.

also, what version of Druid are you running?

**Comment by skilledmonster** [comment]
Application logs? As-in metabase application logs? How can I enable them ?
Or 
Are you asking about broker logs from Druid?

Druid Version # 0.9.0

**Comment by agilliland** [comment]
Metabase application logs.  They are sent to STDOUT wherever your Metabase instance is running.

**Comment by skilledmonster** [comment]
I have been running metabase in the background, is there a way to configure logging so I can write it to a file instead?

**Comment by agilliland** [comment]
What OS are you running on?  If you launched Metabase in a terminal the output should be displaying there.  If you are running it from the .jar on a *nix based system then something like this would work fine for capturing the output in a log file ... `java -jar metabase.jar >> /path/to/log-file`

**Comment by skilledmonster** [comment]
I removed the existing druid database, enabled the logging as suggested above and recreated the database again. I see no errors in the logs. Please see attached log file.
[metabase.txt](https://github.com/metabase/metabase/files/226729/metabase.txt)

**Comment by camsaul** [comment]
errors won't show up in the log file since you're only piping `stdout` to it.

Pipe `stderr` into `stdout` and try it

```bash
java -jar metabase.jar 2>&1 >> /path/to/log-file
```

**Comment by agilliland** [comment]
Cool, thanks @skilledmonster .  So from your logs it doesn't actually look like anything is wrong, which is strange.

It looks like you have 2 tables in your Druid db and for the `Cache22ActivityStats` our driver is only seeing these fields `#{"RATE_COUNT" "timestamp" "count" "TIME_IN_MILLIS"}`, so maybe there is something about that table that is preventing us from seeing the rest of the columns.

I see that there is another table called `Cache22ServerStats`, are you seeing all of the proper columns showing up in that table?  It looks like Metabase found a lot more columns in that table.

**Comment by skilledmonster** [comment]
For Cache22ServerStats, it found lot of columns but they are all metric columns and not dimension columns. 

To give you a context, metabase is only pulling metrics and not dimensions. Druid Schema defines them separately. Check out this link - http://druid.io/docs/latest/ingestion/index.html. Under Data Schema, notice how we specify - dimensionsSpec and metricsSpec. metricsSpec is where we have the numbers and dimensionsSpec is where you can specify your group by columns.
@camsaul - I enabled the other stderr as well and found no errors. 
Hope that helps to investigate further.

**Comment by skilledmonster** [comment]
Do you need any other information to triage this issue?

**Comment by agilliland** [comment]
I suspect that we will need some more information, but I'm not sure what we will need.  Our first goal is to figure out how to reproduce this, which we haven't been able to do yet.

**Comment by skilledmonster** [comment]
If you are have a json druid schema spec that you are using to test Druid functionality with Metabase, please attach it here and I can compare the spec with mine and probably help you with the differences.

**Comment by skilledmonster** [comment]
Folks - Anymore updates on this item? I am waiting on this bug to be resolved so I can completely evaluate my POC project. Let me know if you need any help.

**Comment by salsakran** [comment]
@skilledmonster can you share your data schema with us?

**Comment by skilledmonster** [comment]
@salsakran - I am attaching spec and sample json (data) for the spec.
couple of things - make sure the timestamps in the sample csv file are updated so the rows are within the window period, adjust the firehose location depending on where you are planning to place the csv file. Let me know if you need any other details.

[druid.zip](https://github.com/metabase/metabase/files/237822/druid.zip)

**Comment by skilledmonster** [comment]
@agilliland - does this mean , you found a bug and are planning to release it as part of 0.17.0 milestone release? And when is 0.17.0 milestone scheduled to be released?

**Comment by camsaul** [comment]
no @skilledmonster, we haven't identified any bugs related to this issue yet; we are still looking into it. 0.17 is scheduled to be released within the next week and will include any fixes we were able to make related to this issue.

**Comment by camsaul** [comment]
@skilledmonster I'm taking a look at your schema. What are the results you see if you do the following request?

```
GET http://<broker-host>:<broker-port>/druid/v2/datasources/Cache22activitystats?interval=1900-01-01/2100-01-01
```

**Comment by skilledmonster** [comment]
It is a sample data that I used to load in the past, but should work for you. You can also try using druid.io's standard wikipedia example. I'll try load the data again in the office tomorrow and ping you the results.

But the core issue, is not being able to show Dimension in Metabase which will restrict the group by option. Check out my snapshot from Pivot on how they have Dimensions and Metrics. Dimensions is what is missing in Metabase.

**Comment by skilledmonster** [comment]
I just fired a local instance with some random data and I could get back - 

{"dimensions":["ACTIVITY_NAME","MODULE_NAME","SOURCE","SERVER_TYPE","SERVER_NAME"],"metrics":["count","RATE_COUNT","TIME_IN_MILLIS"]}

**Comment by skilledmonster** [comment]
Any progress guys ?

**Comment by camsaul** [comment]
Not yet @skilledmonster, I'm still looking into it. Something might have changed in 0.9.0 (which just came out a few weeks ago). We're still testing against Druid 0.8.3 but I'm testing your data against both trying to work out the root cause of the issue.

**Comment by skilledmonster** [comment]
Let me know if I can be of any help!

**Comment by skilledmonster** [comment]
oh..this bug got removed from 0.17.0 milestone? I have been waiting on this for a week to submit my evaluation based on the outcome of this issue. Looks like I have to wait much longer?

**Comment by skilledmonster** [comment]
Oh boy! This got removed from 0.17.0 milestone? Let me know if I can be of any help!

**Comment by agilliland** [comment]
Yes, our release comes out today and we were not able to reproduce this so there was no way to fix it.  We'll keep looking into it though.

**Comment by skilledmonster** [comment]
@agilliland - can u point me to your test data n test druid spec used?...i can help you reproduce the isse.

**Comment by agilliland** [comment]
everything is in the codebase https://github.com/metabase/metabase/blob/master/test/metabase/test/data/druid.clj

**Comment by camsaul** [comment]
@skilledmonster are you still seeing this issue?

**Comment by camsaul** [comment]
@skilledmonster ?

**Comment by camsaul** [comment]
@skilledmonster, since we haven't heard anything back about whether you're still seeing this issue since May I'm going to go ahead and close it out. Feel free to open a new issue if you're still having problems.

**Comment by jdelamar** [comment]
Good day. I am actually this exact same issue with Druid 0.10.0 (and 0.10.1-rc2). Is there anything I can do to help fix that?
